src/js/components/search/filters/timePeriod/DateRange.jsx

```diff
@@ -14,35 +15,58 @@ const defaultProps = {
 
 const propTypes = {
     startingTab: React.PropTypes.number,
-    showError: React.PropTypes.func,
     onDateChange: React.PropTypes.func,
-    hideError: React.PropTypes.func,
     startDate: React.PropTypes.object,
     endDate: React.PropTypes.object
 };
 
 export default class DateRange extends React.Component {
 
+    constructor(props) {
+        super(props);
+        this.state = {
+            showError: false,
+            header: '',
+            errorMessage: ''
+        };
+    }
+
+    showError(error, message) {
+        this.setState({
+            showError: true,
+            header: error,
+            errorMessage: message
+        });
+    }
+    hideError() {
+        this.setState({
+            showError: false,
+            header: '',
+            errorMessage: ''
+        });
+    }
+
     render() {
+        let errorDetails = null;
+        if (this.state.showError) {
+            errorDetails = (<DateRangeError
+                header={this.state.header} message={this.state.errorMessage} />);
+        }
         return (
             <div className="dateRangeOption">
                 <div className="date-range-wrapper">
-                    <div className="date-range-wrapper">
-                        <DatePicker
-                            type="startDate" title="Start Date" tabIndex={this.props.startingTab}
-                            onDateChange={this.props.onDateChange}
-                            value={this.props.startDate}
-                            opposite={this.props.endDate} showError={this.props.showError}
-                            hideError={this.props.hideError} />
-                        <div className="through-text">
-                            through
-                        </div>
-                        <DatePicker
-                            type="endDate" title="End Date" tabIndex={this.props.startingTab + 4}
-                            onDateChange={this.props.onDateChange} value={this.props.endDate}
-                            opposite={this.props.startDate} showError={this.props.showError}
-                            hideError={this.props.hideError} />
-                    </div>
+                    <DatePicker
```

Comment:
Can we keep this from happening (datepicker being off screen)? 
![screen shot 2016-11-14 at 11 05 07 am](https://cloud.githubusercontent.com/assets/3594363/20271918/5d1f6bf8-aa5a-11e6-90a5-b035d58e5322.png)